rust/pact_ffi/src/verifier/mod.rs

```diff
@@ -147,6 +147,46 @@ ffi_fn! {
     }
 }
 
+ffi_fn! {
+    /// Set the verification options for the Pact verifier.
+    ///
+    /// `publish` is a boolean value. Set it to greater than zero to turn the option on.
+    /// `disable_ssl_verification` is a boolean value. Set it to greater than zero to turn the option on.
+    ///
+    /// # Safety
+    ///
+    /// All string fields must contain valid UTF-8. Invalid UTF-8
+    /// will be replaced with U+FFFD REPLACEMENT CHARACTER.
+    ///
+    fn pactffi_verifier_set_verification_options(
+      handle: *mut handle::VerifierHandle,
+      publish: c_uchar,
+      provider_version: *const c_char,
+      build_url: *const c_char,
+      disable_ssl_verification: c_uchar,
+      request_timeout: c_ulong,
+      provider_tags: *const *const c_char,
+      provider_tags_len: c_ushort
+    ) {
+      let handle = as_mut!(handle);
+      let provider_version = safe_str!(provider_version);
+      let build_url = safe_str!(build_url);
```

Comment:
I tested with `provider_version = null` or `build_url = null`, it stop right here, not running further.

Is it okay? If not, I think we must use `if_null`. Can you suggest default values for `provider_version` and `build_url` if so?

Same for `url` in `pactffi_verifier_set_provider_state`

Reply:
`provider_version` is required, so if that is not set, we should error at that point.

`build_url` is optional, so we should check for null and set a default then. 

For the provider state url, I think that can be optional. The provider states will just be skipped then.

Reply:
Just checked the `safe_str` macro, and it will panic if the pointer is null, which is probably not the behaviour we want.

Reply:
Use the `if_null` function for `build_url`, and set the default to the empty string.

For `provider_version`, it is mandatory, so you need to add an error block as a second block that returns an error code when the `safe_str!` panics. See `pactffi_message_set_description` for an example on how to do that.

Reply:
Thanks. I learnt something new. Updated.